packages/nimble-components/src/table/specs/table-column-specs/table-column-text-field.md

```diff
@@ -0,0 +1,175 @@
+-   [`TableCellState`](../table-columns-hld.md#tablecellstate-interface)
+
+The `TableColumnTextField` will extend the `TableColumn` in a manner similar to the following:
+
+```TS
+type TableColumnTextCellData = StringField<'value'>;
+type TableColumnTextColumnConfig = { placeholder: string };
+
+public class TableColumnTextField extends TableColumn<TableColumnTextCellData, TableColumnTextColumnConfig> {
+    ...
+
+    @attr({ attribute: 'value-key'})
+    public valueKey: string;
+
+    @attr
+    public placeholder: string; // Column auxiliary configuration
+
+    public getRowDataKeys(): string[] {
+        return [valueKey];
+    }
+
+    ...
+}
+```
+
+### Anatomy
+
+As this component has no template, there is no anatomy of concern.
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component.
+
+### Visual Appearance
+
+The visual appearance of the text content will match that of a `nimble-text-field`.
+
+---
+
+## Implementation
+
+For the `cellTemplate` implementation required for a `TableColumn<>` implementation we will provide something similar to the following:
+
+```TS
+public class TableColumnTextField ...
+{
+    ...
+
+    public cellDataKeyNames = ['value'] as const;
+
+    public readonly cellStyles: ElementStyles => css`
+        .text-value {
+            // set necessary text-value styles
+        }
+
+        .placeholder {
+            // set necessary placeholder styles
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>> =>
+        html<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>>`
+            ${when(x => x.data.value, html`
+                <span class="text-value">{x => x.data.value}</span>
+            `)}
+            ${when(x => !x.data.value, html`
+                <span class="placeholder">{x => x.columnConfig.placeholder}</span>
+            `)}
+        `;
+
+    ...
+}
+```
+
+### Alternatives considered
+
+We are using `span` elements for the text rendering of the values instead of a `nimble-text-field` for performance's sake, as we avoid the notably heavier cost of using a custom element. The other benefits that using a `nimble-text-field` offer, such as built-in styling and a placeholder implementation seem trivial to replicate, which seems worth it for a performance improvement.
+
+One notable difference in behavior is that the proposed implementation will not support a behavior present in the `nimble-text-field` where a user can begin dragging at an arbitrary location in the text, and go to the end of the text, even if it has been clipped by the column (and thus showing an `...`). Users will still be able to double-click such text and copy the entire contents.
```

Comment:
I'm also curious what general text styling we plan to support. Can clients override fonts and styles (e.g. set a column/cell to be italic)? [Control the ellipsis location](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow)? Are the classes listed in your implementation (`placeholder` and `text-value`) considered part of the public API (I'm guessing not since they're in the shadow DOM?)?

Reply:
At the moment, I'm not sure what the best approach will be to provide custom styling mechanisms to the column content. For columns where we want to provide that level of customization, it might have to be done through its top-level API, providing individual configurable attributes.

You are correct, though, in that the classes you mentioned are not considered a part of the public API, nor do I think we would want them to be (as I'm not sure the client could even use them properly from that scope).

Would you like for me to explicitly mark this concern as "out-of-scope" in the 'Non-goals' section? I think we can revisit this later if the demand is there.

Reply:
Yea I think custom styling should be marked out of scope and something we revisit when use-cases come up. In general our philosophy is to be opinionated on styling so we should be thoughtful about how that is exposed.

Reply:
Do we have known use-cases?

Reply:
Marking it as out of scope for now sounds good to me. I can't think of any SLE use cases for custom styling off the top of my head. I can imagine use cases like setting the text or background color based on value (e.g. negative is red) or preferring text truncation at different locations depending on the type of string being displayed. But consulting with clients and UX and then gathering these into an issue and prioritizing it sound like jobs for @nate-ni!

Reply:
Went ahead and added this as a non-goal.